components/clarity-vscode/README.md

```diff
@@ -1,50 +1,69 @@
-# Clarity for VSCode
+# Clarity for Visual Studio Code
 
-Clarity is a decidable smart contract language that optimizes for predictability and security. Smart contracts allow developers to encode essential business logic on a blockchain.
+Clarity is a **decidable** smart contract language that optimizes for predictability and security. Developed by Hiro, Clarity enables smart contract developers to encode essential business logic on a blockchain.
 
-This extension aims to help developer write safe and clean Clarity code, it includes syntax highlighting, auto-completion, linting, debugging and [safety checks](https://www.hiro.so/blog/new-safety-checks-in-clarinet),.
+A programming language is referred to as **decidable** if you can know with certainty, from the code itself, what the program will do. Clarity is was designed to be intentionally Turing incomplete, thereby avoiding the problem of `Turing complexity` (the number of steps it takes for a turing machine to run an algorithm on a given input size). This allows for complete static analysis of the entire call graph of a given smart contract, while Hiro's support for types and type checkers can eliminate whole classes of bugs like unintended casts, reentrancy bugs, and reads of uninitialized values.
 
-## Support for [vscode.dev](https://vscode.dev)
+The Language Server Protocol (LSP) defines the protocol used between an editor or IDE, and a language server that provides language features such as auto-complete, go-to-definition, find-all-references, and others.
 
-The Clarity extension runs in the browser so it can be used with [vscode.dev](https://vscode.dev) and [github.dev](https://github.dev), the same way it runs in VSCode for Desktop.
+This Clarity-LSP project aims at leveraging the decidability quality of Clarity and the LSP to provide insights about your code, without publishing your smart contracts to a blockchain, while also enabling you to run the Clarity for Visual Studio Code extenstion in-browser.
 
-The Debugger is not implemented yet on the browser.
+![screenshot](images/screenshot.png)
 
-## Debugging
+## Clarity for Visual Studio Installation
 
-The debugger included in this extension allows to run Clarity code line by line which is a powerful tool to understand what precisely happens. Read this [blog post](https://www.hiro.so/blog/how-to-debug-your-smart-contracts-with-clarinet) to know more about Clarity debugging.
+You can install the latest release of the plugin from the [marketplace](https://marketplace.visualstudio.com/items?itemName=hirosystems.clarity-lsp).
+## Features
 
-In order to use the Debugger (DAP), this extension relies on a local installation of Clarinet. To install Clarinet, please follow the instructions [here](https://github.com/hirosystems/clarinet#installation).
+The features described below are available when you install the Clarity for Visual Studio plugin.
 
----
-## Contributes
```

Comment:
I think we should keep this "Contributing" part

Reply:
Added this back into the README.